src/org/opensolaris/opengrok/search/SearchEngine.java

```diff
@@ -377,9 +410,32 @@ private int search(List<Project> projects, File root) {
             }
         }
         int count = hits == null ? 0 : hits.length;
+        queryBuilder = newBuilder;
         return count;
     }
 
+    /**
+     * Gets the queried score docs from {@code search(...)} if it was called.
+     * @return a defined instance if a query succeeded, or {@code null}
+     */
+    public ScoreDoc[] scoreDocs() {
+        return hits;
+    }
+
+    /**
+     * Gets the document of the specified {@code docId} from
+     * {@code search(...)} if it was called.
+     * @param docId
```

Comment:
Could you please add comment to the `@param` and `@throws` tags? :) Or remove them if you prefer to.